src/screens/Resources/HomeScreen.js

```diff
@@ -0,0 +1,167 @@
+import { collections, nav } from './variables';
+
+export default function HomeScreen(props) {
+  const { navigation } = props;
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [loading, setLoading] = useState(false);
+  const [snapshot, setSnapshot] = useState(null);
+  const [initialLoad, setInitialLoad] = useState(true);
+
+  const loadScreen = useCallback((id, title) => {
+    setLoading(true);
+    firestore()
+      .collection(collections.subcategories)
+      .where('parent', '==', id)
+      .get()
+      .then((subcategorySnapshot) => {
+        if (!subcategorySnapshot.empty) {
+          navigation.push(nav.subcategories, { snapshot: subcategorySnapshot, header: title });
+        } else {
+          firestore()
+            .collection(collections.items)
+            .where('parent', '==', id)
+            .get()
+            .then((itemSnapshot) => {
+              navigation.push(nav.itemList, { snapshot: itemSnapshot, header: title });
+            })
+            .catch((error) => {
+              setErrorMessage(error.message);
+              setLoading(false);
+            });
+        }
+      })
+      .catch((error) => {
+        setErrorMessage(error.message);
+        setLoading(false);
+      });
+  }, [navigation]);
+
+  useEffect(() => {
+    firestore()
+      .collection(collections.categories)
+      .get()
+      .then((querySnapshot) => {
+        setSnapshot(querySnapshot);
+        setInitialLoad(false);
+      })
+      .catch((error) => {
+        setErrorMessage(error.message);
+        setInitialLoad(false);
+      });
+  }, []);
+
+  function handleSignOut() {
+    auth()
+      .signOut()
+      .then(() => {
+        navigation.navigate('Auth');
+      }).catch((error) => {
+        setErrorMessage(error.message);
+      });
+  }
+
+  return (
+    <ScrollView contentContainerStyle={HomeStyles.scrollviewContainer}>
+      <NavigationEvents
+        onWillFocus={() => {
+          setLoading(false);
+        }}
+      />
+      {errorMessage
+        && (
+        <Text style={{ color: 'red' }}>
+          {errorMessage}
+        </Text>
+        )}
+      { (loading || initialLoad)
+        && (
+        <ActivityIndicator size="large" />
+        )}
+      <View style={HomeStyles.categoryButtonView}>
+        { !initialLoad
+          && (snapshot.docs.map((doc) => {
+            const {
+              title, thumbnail,
+            } = doc.data();
+
+            return (
+              <TouchableOpacity
+                style={HomeStyles.categoryButton}
+                key={doc.id}
+                disabled={loading}
+                onPress={() => loadScreen(doc.id, title)}
+              >
+                <Image source={{ uri: thumbnail.src }} style={HomeStyles.categoryImage} />
+                <Text style={HomeStyles.categoryText}>
+                  {`${title}`}
+                </Text>
+              </TouchableOpacity>
+            );
+          }))}
+      </View>
```

Comment:
Might be worthwhile to create a separate component for this list so this component is less monolithic.